### PR TITLE
Adds support for appending values to attributes

### PR DESCRIPTION
**Pull request by julianschiavo**
This PR adds support for `append(additionalValue:separator:)` on attributes to allow adding additional values to an attribute. An example use case for this would be adding additional classes.

Also adds a new test, `testAppendingBodyCSSClass()` that checks that the value is appended correctly.

**Example Code:**
```swift
.body(Node.class("a").append("b"))
// Output: <body class="a b"></body>
```

Note: I'm not sure if there's a better way but due to the addition of `.append("b")` it requires adding `Node.` to the start of the attribute otherwise Xcode throws an error.

### CODE REVIEW

File: Tests/PlotTests/HTMLTests.swift [hunk 1]
```diff
@@ -189,6 +189,11 @@ final class HTMLTests: XCTestCase {
         let html = HTML(.body(.class("a"), .class("b")))
         assertEqualHTMLContent(html, #"<body class="b"></body>"#)
     }
+    
+    func testAppendingBodyCSSClass() {
+        let html = HTML(.body(Node.class("a").append("b")))
```
Comment (JohnSundell) [inline Tests/PlotTests/HTMLTests.swift]: I get what you were going for with this API design, but the problem is that it sort of "breaks" the DSL, by forcing you to refer to `Node` explicitly. Ideally, the end user should never have to worry much about the `Node` type, and instead be able to just add elements and attributes purely using dot-syntax. So I think it'd be better to enable this to be expressed as `.body(.append(.class("a")))` instead.

**Comment by john-mueller** [comment]
Could we instead let nodes that can take multiple values accept a collection? Something like:

```swift
.body(.class(["a", “b”]))
// Output: <body class="a b"></body>
```

Perhaps via a protocol with a default implementation?

**Comment by john-mueller** [comment]
Oh, I forgot that each `Attribute`/`Node` is implemented as a static function, so I don't think there's a protocol solution.

Maybe we could change the appropriate attributes to take a variadic:

```swift
static func `class`(_ classNames: String...) -> Node {
    .attribute(named: "class", value: classNames.joined(separator: " "))
}
```

Then you could just call

```swift
.body(.class("a", “b”))
// Output: <body class="a b"></body>
```

**Comment by julianschiavo** [comment]
> Oh, I forgot that each `Attribute`/`Node` is implemented as a static function, so I don't think there's a protocol solution.
> 
> Maybe we could change the appropriate attributes to take a variadic:
> 
> ```swift
> static func `class`(_ classNames: String...) -> Node {
>     .attribute(named: "class", value: classNames.joined(separator: " "))
> }
> ```
> 
> Then you could just call
> 
> ```swift
> .body(.class("a", “b”))
> // Output: <body class="a b"></body>
> ```
> 
> What do you think @julianschiavo, @JohnSundell?

Yeah, I think this may be easier to implement and sustain in the future then adding a completely separate `.append` function. Let's see what @JohnSundell says before I make any changes either way.

**Comment by JohnSundell** [comment]
Perhaps I misunderstood something from our Twitter conversation @julianschiavo, but wasn't the point of this to enable you to append a value to an existing attribute from multiple call sites? If you can do everything at the same call site, then can't you just space-separate the classes using the existing API?

```swift
.body(.class("one two three"))
```

I see very limited value of adding either an Array- or variadic-based overload just to join a few strings by space, to be honest.

**Comment by julianschiavo** [comment]
Yeah, sorry I was just considering the different options and in testing this on a personal project I have found that my current implementation on this PR does not work well too. I agree that it would make more sense with your idea, I'm just not sure what it would take. I'll take a look nonetheless and see what I come up with.

**Comment by julianschiavo** [comment]
I just pushed a remake that uses the API style suggested by @JohnSundell initially, where it's now used as such:

```swift
.body(.class("a"), .append(.class("b")))
// Output: <body class="a b"></body>
```

Let me know what you think and if any changes are needed.

**Comment by julianschiavo** [comment]
Hey @JohnSundell, has there been any update on this? Thanks!

**Comment by JohnSundell** [comment]
Sorry for the very late reply on this. The good news, though, is that this use case is now supported thanks to the new Component API. The behavior of appending classes is even the default when using that new API. For example, here's how two classes could be added to a `<div>` element:

```swift
Div {
    // Content
    ...
}
.class("one")
.class("two")
```

Which produces the following HTML:

```html
<div class="one two">...</div>
```

Since `Node` also conforms to the new `Component` protocol, you can also get the exact same behavior when using `Node`-based elements as well, by applying the `class` modifier to such nodes. The only caveat is that you'd then have to wrap the returned result in a `.component` node to convert  it back to a `Node`, for example like this:

```swift
.div(
    .component(
        Node.div(
            // Content
            ...
        )
        .class("one")
        .class("two")
    )
)
```

Which produces the following HTML:

```html
<div><div class="one two">...</div></div>
```

Thanks a lot for all of your work on this @julianschiavo, but I still think that the previous behavior is the right one for `Node`-based hierarchies (that is, enable the user to override classes and other attributes by assigning new ones), while the append behavior makes perfect sense in this new `Component`-based world. So hopefully these new changes that were introduced in Plot `0.9.0` will let you achieve what you were looking to achieve. I'm going to close this PR, but once again, thanks for contributing, and sorry for taking so long to reply to this.

**Pull request closed without merging**